pkg/process/metadata/parser/service.go

```diff
@@ -295,7 +297,18 @@ func parseCommandContextJava(args []string) string {
 
 			if arg = trimColonRight(arg); isRuneLetterAt(arg, 0) {
 				if strings.HasSuffix(arg, javaJarExtension) {
-					return arg[:len(arg)-len(javaJarExtension)]
+					jarName := arg[:len(arg)-len(javaJarExtension)]
+					if !strings.HasSuffix(jarName, javaSnapshotSuffix) {
+						return jarName
+					}
+					jarName = jarName[:len(jarName)-len(javaSnapshotSuffix)]
+					components := strings.Split(jarName, "-")
```

Comment:
May save on some allocations by using `strings.LastIndex(jarName, "-")`; this will also avoid use of the `strings.Join` below.